### PR TITLE
more updates for OpenStack

### PR DESCRIPTION
**Pull request by jjasghar**
Here are some updates that I had to make to the cookbook.

### CODE REVIEW

**Comment by jjasghar** [comment]
Not quite done yet though, i'm hitting the following error(s)

```bash
22:09:01 ~ > ssh openstack-user@172.16.20.61

________/\\\\\\\\\__/\\\________/\\\__/\\\\\\\\\\\\\\\__/\\\\\\\\\\\\\\\_
 _____/\\\////////__\/\\\_______\/\\\_\/\\\///////////__\/\\\///////////__
  ___/\\\/___________\/\\\_______\/\\\_\/\\\_____________\/\\\_____________
   __/\\\_____________\/\\\\\\\\\\\\\\\_\/\\\\\\\\\\\_____\/\\\\\\\\\\\_____
    _\/\\\_____________\/\\\/////////\\\_\/\\\///////______\/\\\///////______
     _\//\\\____________\/\\\_______\/\\\_\/\\\_____________\/\\\_____________
      __\///\\\__________\/\\\_______\/\\\_\/\\\_____________\/\\\_____________
       ____\////\\\\\\\\\_\/\\\_______\/\\\_\/\\\\\\\\\\\\\\\_\/\\\_____________
        _______\/////////__\///________\///__\///////////////__\///______________


Welcome to the Chef Server Marketplace Appliance!

If this your first time logging in you'll need to set up the Chef Server by
running the following command:

sudo chef-server-ctl marketplace-setup

If you need help, you can find documentation at https://docs.chef.io/openstack_marketplace.html

You can also contact our Support Team via email at openstack@chef.io


[openstack-user@aoeu ~]$ curl https://172.16.20.61
curl: (7) couldn't connect to host
[openstack-user@aoeu ~]$ curl https://127.0.0.1
curl: (7) couldn't connect to host
[openstack-user@aoeu ~]$ getenforce
Enforcing
[openstack-user@aoeu ~]$ sudo setenforce Permissive
[openstack-user@aoeu ~]$ sudo chef-server-ctl test
Configuring logging...
Creating platform...
Configured URL: https://
Creating org pedant_testorg_aoeu_2009
/opt/opscode/embedded/service/oc-chef-pedant/lib/pedant/core_ext/net_http.rb:22:in `initialize': no implicit conversion of nil into String (TypeError)
	from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant/core_ext/net_http.rb:22:in `open'
	from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant/core_ext/net_http.rb:22:in `block in connect'
	from /opt/opscode/embedded/lib/ruby/2.1.0/timeout.rb:76:in `timeout'
	from /opt/opscode/embedded/lib/ruby/2.1.0/timeout.rb:127:in `timeout'
	from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant/core_ext/net_http.rb:22:in `connect'
	from /opt/opscode/embedded/lib/ruby/2.1.0/net/http.rb:863:in `do_start'
	from /opt/opscode/embedded/lib/ruby/2.1.0/net/http.rb:852:in `start'
	from /opt/opscode/embedded/service/gem/ruby/2.1.0/bundler/gems/rest-client-ba0d12258b77/lib/restclient/request.rb:183:in `transmit'
	from /opt/opscode/embedded/service/gem/ruby/2.1.0/bundler/gems/rest-client-ba0d12258b77/lib/restclient/request.rb:69:in `execute'
	from /opt/opscode/embedded/service/gem/ruby/2.1.0/bundler/gems/rest-client-ba0d12258b77/lib/restclient/request.rb:36:in `execute'
	from /opt/opscode/embedded/service/gem/ruby/2.1.0/bundler/gems/rest-client-ba0d12258b77/lib/restclient.rb:73:in `post'
	from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant/request.rb:143:in `authenticated_request'
	from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant/request.rb:165:in `post'
	from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant/platform.rb:322:in `block in create_org'
	from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant/platform.rb:321:in `times'
	from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant/platform.rb:321:in `create_org'
	from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant/platform.rb:123:in `org_from_config'
	from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant/platform.rb:42:in `initialize'
	from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant.rb:79:in `new'
	from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant.rb:79:in `create_platform'
	from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant.rb:58:in `setup'
	from ./bin/oc-chef-pedant:10:in `<main>'
[openstack-user@aoeu ~]$
```

I also think we should automate: `[openstack-user@aoeu ~]$ sudo setenforce Permissive` according to the main docs we have to set it that anyway.

File: attributes/default.rb [hunk 1]
```diff
@@ -6,6 +6,7 @@
 default['marketplace_image']['analytics_version'] = 'latest'
 default['marketplace_image']['license_count'] = 5
 default['marketplace_image']['support_email'] = 'aws@chef.io'
+default['marketplace_image']['marketplace_URL'] = 'https://docs.chef.io/aws_marketplace.html'
```
Comment (ryancragun) [inline attributes/default.rb]: url should probably be lower case

File: recipes/_bootstrap_server.rb [hunk 2]
```diff
@@ -73,7 +73,10 @@
 motd '50-chef-marketplace-appliance' do
   source 'server-motd.erb'
   cookbook 'marketplace_image'
-  variables support_email: node['marketplace_image']['support_email']
+  variables({
+              support_email: node['marketplace_image']['support_email'],
+              marketplace_URL: node['marketplace_image']['marketplace_URL']
```
Comment (ryancragun) [inline recipes/_bootstrap_server.rb]: s/URL/url/

File: recipes/_openstack_security.rb [hunk 3]
```diff
@@ -24,9 +24,5 @@
       expect(user('openstack-user')).to have_home_directory('/home/openstack-user')
       expect(user('openstack-user')).to have_login_shell('/bin/bash')
     end
-
-    it 'does not have a password' do
-      expect(command('passwd -S openstack-user').stdout).to match(/Password locked/)
-    end
```
Comment (ryancragun) [inline recipes/_openstack_security.rb]: I don't think that giving a default password to a user on a public image is a good thing.  I think a locked password, combined with sudo access and SSH should be just fine.

File: templates/default/server-motd.erb [hunk 4]
```diff
@@ -22,7 +22,7 @@ running the following command:
 
 sudo chef-server-ctl marketplace-setup
 
-If you need help, you can find documentation at https://docs.chef.io/aws_marketplace.html
+If you need help, you can find documentation at <%= @marketplace_URL %>
```
Comment (ryancragun) [inline templates/default/server-motd.erb]: s/URL/url/

File: templates/default/openstack-cloud-init.erb [hunk 5]
```diff
@@ -7,6 +7,8 @@ ssh_pwauth: false
 locale_configfile: /etc/sysconfig/i18n
 mount_default_fields: [~, ~, 'auto', 'defaults,nofail', '0', '2']
 
+manage_etc_hosts: true
```
Comment (ryancragun) [inline templates/default/openstack-cloud-init.erb]: Are we sure that we want to regenerate it on reboots?  Perhaps setting this to [localhost](http://bazaar.launchpad.net/~cloud-init-dev/cloud-init/trunk/view/head:/doc/examples/cloud-config.txt#L470) is a better option?

File: templates/default/http.erb [hunk 6]
```diff
@@ -0,0 +1,3 @@
+# Port 80 for http
```
Comment (ryancragun) [inline templates/default/http.erb]: Can we name this something a bit more specific?

File: templates/default/https.erb [hunk 7]
```diff
@@ -0,0 +1,3 @@
+# Port 443 for https
```
Comment (ryancragun) [inline templates/default/https.erb]: Perhaps rename this template as well?

File: templates/default/ssh.erb [hunk 8]
```diff
@@ -0,0 +1,3 @@
+# Port 22 for ssh
```
Comment (ryancragun) [inline templates/default/ssh.erb]: This one too!

**Comment by ryancragun** [comment]
Yeah, after discussions I think it best to not introduce iptables into the mix.

**Pull request merged**